### DISCUSSION TITLE
Looking for info on colored coins

### DISCUSSION THREAD

**Discussion by SlugFiller** [Chia-Network/chia-blockchain#7971] [Q&A]
So, Chia supposedly supports colored coins. But the info about it seems to be limited to an old blog post saying it exists, and an old video kind-of explaining it using wallet software that looks nothing like the current version. Both give explanations that leave a lot to be desired.

In particular, the demo seems to imply that a color coin can only be created once, with a limited supply. This is somewhat counter-productive, even if you can technically apply 1 mojo=1 coin, because a standard theoretical use case, like selling, say, tickets to a service, require being able to create as many coins as there are customers.

The blog, on the other hand, claims that creating new coins can be done as part of an "offer", while the video only shows making a simple trade offer of colored coins for XCH. There's also no information on whether the "offer" is for a single trade, or is reusable for multiple trades. I also read something about being able to "cancel" an offer. That would be pretty important if, for instance, one wants to raise the price of a coin, to prevent using the previous, cheaper offer.

So, I want to know what colored coins are actually capable of, if it can be done with the current Chia client, and how. I'll focus on a specific use case: Service tickets. A nearly equivalent use case would be stablecoins, where the "service" is getting an equivalent amount of fiat in exchange for the tickets.

- Service tickets should be infinitely producible to match the customer count.
- Only the service provider should have permission to produce more tickets.
- Tickets should be indivisible. You can't have half a ticket. In essence, it should be equivalent to 1 mojo.
- Tickets should be indistinguishable. The service provider shouldn't need to keep a list of all produced tickets. Rather, two tickets to the same service should be equivalent.
- If possible, producing a ticket should be part of an "offer", with the service provider receiving a certain compensation, minus a mojo fee for the ticket, and the ticket being produced and delivered to the payer.
- It must be possible to "edit" an offer to a higher price, "cancelling" the previous offer from being usable.
- Tickets must be verifiable - The service provider must be able to verify that a ticket is present in a certain wallet, and that it is a valid ticket created by the service provider, or by a valid offer created by the service provider.

So, is any of this possible? And if so, how?

**Comment by hoffmang9** [comment]
Our coloured coin implementation is in the repository and ships with Chia but we temporarily disabled the UI as we want to make some enhancements. You can see the (simple) implementation here: https://github.com/Chia-Network/chia-blockchain/tree/main/chia/wallet/cc_wallet

You should check out our Chialisp training and possibly attend one of the Q&A sessions. https://chialisp.com/training/

All of what you want to do is possible. Enhancing our reference coloured coin implementation is one of the very next development priorities now that we've updated the Chialisp training and documentation.
